server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -168,6 +169,37 @@ public void execute(org.quartz.JobExecutionContext context) throws org.quartz.Jo
                 throw new org.quartz.JobExecutionException(errmsg, e);
             }
         }
+
+        @Override
+        public String getName() {
+            return this.getClass().getSimpleName();
+        }
+
+        @Override
+        public void triggerComplete(Trigger trigger, org.quartz.JobExecutionContext context,
+            Trigger.CompletedExecutionInstruction triggerInstructionCode) {
+            // Intentionally left empty
+        }
+
+        @Override
+        public void triggerFired(Trigger trigger, org.quartz.JobExecutionContext context) {
+            // Intentionally left empty
+        }
+
+        @Override
+        public void triggerMisfired(Trigger trigger) {
+            String jobKey = trigger.getKey().getName();
+
+            log.warn("Trigger misfired for job: {} [start: {}, end: {}, next fire time: {}, " +
+                "final fire time: {}, priority: {}, misfire instruction: {}]",
+                jobKey, trigger.getStartTime(), trigger.getEndTime(), trigger.getNextFireTime(),
+                trigger.getFinalFireTime(), trigger.getPriority(), trigger.getMisfireInstruction());
+        }
+
+        @Override
+        public boolean vetoJobExecution(Trigger trigger, org.quartz.JobExecutionContext context) {
+            return false;
```

Comment:
Do you think we should add a condition to check if Candlepin is in SUSPEND mode and not allow job to be executed.

Reply:
Nah. Suspend mode is already handled by pausing both the scheduler and the message receiver. We don't need explicit handling here.